src/main/java/pages/PageObject.java

```diff
@@ -1,15 +1,39 @@
 package pages;
 
+import hooks.Hooks;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.PageFactory;
 
 public abstract class PageObject {
 
-    protected WebDriver driver;
+    private WebDriver driver = Hooks.getDriver();
+    private HomePage homePage;
+    private ProductPage productPage;
+    private ShoppingCart shoppingCart;
+    private ProductDetailsPage productDetailsPage;
```

Comment:
revert this, PageObject class is an abstract class that must contain just common staff for all page objects